cmd/build/v2/hasher.go

```diff
@@ -105,15 +102,13 @@ func (sh serviceHasher) hash(buildInfo *model.BuildInfo, commitType, commitHash
 	// We use a builder to avoid allocations when building the string
 	var b strings.Builder
 
-	fmt.Fprintf(&b, "%s:%s;", commitType, commitHash)
+	fmt.Fprintf(&b, "commit:%s;", commitHash)
 	fmt.Fprintf(&b, "target:%s;", buildInfo.Target)
 	fmt.Fprintf(&b, "build_args:%s;", argsText)
 	fmt.Fprintf(&b, "secrets:%s;", secretsText)
 	fmt.Fprintf(&b, "context:%s;", buildInfo.Context)
 	fmt.Fprintf(&b, "dockerfile:%s;", buildInfo.Dockerfile)
-	if commitType == buildContextCommitType {
-		fmt.Fprintf(&b, "dockerfile_content:%s;", getDockerfileContent(buildInfo.Dockerfile))
-	}
+	fmt.Fprintf(&b, "dockerfile_content:%s;", getDockerfileContent(buildInfo.Dockerfile))
```

Comment:
while validating with `okteto/movies` if I run `okteto build -l=debug`, I've noticed that this tries to always open `Dockerfile` which retults in:

```
INFO[0001] error trying to read Dockerfile: %wopen Dockerfile: no such file or directory
```

Instead, if should try reading the content of the `worker/Dockerfile`, etc..

Can you have a look as well?

Reply:
Added on https://github.com/okteto/okteto/pull/4083/commits/8b112bbf068c337d9e55782a7996f6d873ddd5a5